From fe5ab69b9789840a67fa5013c431cf1b005b1e10 Mon Sep 17 00:00:00 2001
From: Durgadoss R <durgadoss.r@intel.com>
Date: Wed, 30 Sep 2015 17:42:23 +0530
Subject: [PATCH] SQUASHME! [VPG]: drm/i915: Save and restore link values

During upfront_link_training, existing values of lane_count
and link_bw are modified to train at various levels. Once
link training is done, the old values (that were used before
starting the upfront_link_train) should be restored to the
variables in intel_dp structure. Only the value in
intel_dp->dpcd array should be retained as modified
during upfront training.

SQUASHME! - This patch should be squashed into the following existing patch:
Author:    Durgadoss R <durgadoss.r@intel.com>
Date:      Fri, 22 May 2015 14:30:07 +0530
Change-Id: I2ae866b32a0998abcf32826efa2504e203dc8ffa
Subject:   FOR_UPSTREAM [VPG]: drm/i915: Enable Upfront link training
           for type-C DP support

Change-Id: I05e19b02b0cd05925c5de6aab60122fe4761b859
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-4819
Signed-off-by: Durgadoss R <durgadoss.r@intel.com>
---
 drivers/gpu/drm/i915/intel_display.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/gpu/drm/i915/intel_display.c b/drivers/gpu/drm/i915/intel_display.c
index a737c6e..31b4d99 100644
--- a/drivers/gpu/drm/i915/intel_display.c
+++ b/drivers/gpu/drm/i915/intel_display.c
@@ -14996,6 +14996,7 @@ bool chv_upfront_link_train(struct drm_device *dev,
 	struct intel_encoder *encoder = connector->encoder;
 	bool found = false;
 	bool valid_crtc = false;
+	uint8_t tmp_lane_count, tmp_link_bw;
 
 	if (!connector || !encoder) {
 		DRM_DEBUG_KMS("dp connector/encoder is NULL\n");
@@ -15036,6 +15037,10 @@ start_link_train:
 					pipe_name(crtc->pipe));
 	found = false;
 
+	/* Save the existing lane_count and link_bw values */
+	tmp_lane_count = intel_dp->lane_count;
+	tmp_link_bw = intel_dp->link_bw;
+
 	/* Initialize with Max Link rate & lane count supported by panel */
 	intel_dp->link_bw =  intel_dp->dpcd[DP_MAX_LINK_RATE];
 	intel_dp->lane_count = intel_dp->dpcd[DP_MAX_LANE_COUNT] &
@@ -15103,6 +15108,11 @@ exit:
 	if (found)
 		DRM_DEBUG_KMS("upfront link training passed. lanes:%d bw:%d\n",
 				intel_dp->lane_count, intel_dp->link_bw);
+
+	/* Restore lane_count and link_bw values */
+	intel_dp->lane_count = tmp_lane_count;
+	intel_dp->link_bw = tmp_link_bw;
+
 	return found;
 }
 
-- 
1.9.1

